From fb2f7284df040f6dc97445ab7f49a6bc93d082f2 Mon Sep 17 00:00:00 2001
From: Dmitry Ornatsky <cdo021c@motorola.com>
Date: Wed, 20 Feb 2013 15:11:39 -0600
Subject: [PATCH 037/959] IKJB42MAIN-4263 drivers: misc: add utag access to SKU
 and Date

Modify utags module to display SKU and Date in /proc/config.
Sync utag id definitions with bootloader.

Change-Id: Ie4bb9536dac358cef9c72bead8faab2dce470654
Signed-off-by: Dmitry Ornatsky <cdo021c@motorola.com>
Reviewed-on: http://gerrit.pcs.mot.com/519720
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Check Patch <CHEKPACH@motorola.com>
Reviewed-by: Klocwork kwcheck <klocwork-kwcheck@sourceforge.mot.com>
Reviewed-by: Jeffrey Carlyle <jeff.carlyle@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
(cherry picked from commit 408ccb24e5e15b3b0c7c1222aef9ee20870ef2d2)
---
 drivers/misc/utag/utags.c |  2 ++
 drivers/misc/utag/utags.h | 40 ++++++++++++++++++++++++++++++++++++++++
 2 files changed, 42 insertions(+)

diff --git a/drivers/misc/utag/utags.c b/drivers/misc/utag/utags.c
index 3db12005297..f53c4b2d307 100644
--- a/drivers/misc/utag/utags.c
+++ b/drivers/misc/utag/utags.c
@@ -55,6 +55,8 @@ static struct utag_node utag_tab[] = {
 	{UTAG_BATTERY, "battery", NULL, NULL, NULL, NULL},
 	{UTAG_DLOAD_MODE, "dload", NULL, NULL, NULL, NULL},
 	{UTAG_IMEI, "imei", NULL, NULL, NULL, NULL},
+	{UTAG_SKU, "sku", NULL, NULL, NULL, NULL},
+	{UTAG_DATE, "date", NULL, NULL, NULL, NULL},
 	{UTAG_END, "all", NULL, NULL, NULL, NULL},
 	{0, "", NULL, NULL, NULL, NULL},
 };
diff --git a/drivers/misc/utag/utags.h b/drivers/misc/utag/utags.h
index 2da61899536..31af8d095d0 100644
--- a/drivers/misc/utag/utags.h
+++ b/drivers/misc/utag/utags.h
@@ -191,6 +191,46 @@ struct frozen_utag {
  */
 #define UTAG_IMEI  0xCAFE0016
 
+/*
+ * CMDLINE OVERLAY: kernel command line parameters which
+ * are appended to the command line
+ * or substitute existing parameters
+ */
+#define UTAG_CMDLINE_OVERLAY  0xCAFE0017
+
+/*
+ * TOD offset: offset in seconds of current time of day
+ * against PMIC RTC
+ * (offset + rtc) == seconds since Epoch (Jan. 1 1970 UTC 00:00AM)
+ */
+#define UTAG_TOD_OFFSET  0xCAFE0018
+
+/*
+ * SKU: Stock Keeping Unit code
+ */
+#define UTAG_SKU  0xCAFE0019
+
+/*
+ * Date Code: mm-dd-yyyy
+ */
+#define UTAG_DATE  0xCAFE001A
+
+/*
+ * tags to store android properties
+ */
+#define UTAG_BUILD_DATE           0xCAFE1000
+#define UTAG_BUILD_ID             0xCAFE1001
+#define UTAG_BUILD_TAGS           0xCAFE1002
+#define UTAG_BUILD_TYPE           0xCAFE1003
+#define UTAG_BUILD_USER           0xCAFE1004
+#define UTAG_BUILD_VER_CODENAME   0xCAFE1005
+#define UTAG_BUILD_VER_INC        0xCAFE1006
+#define UTAG_BUILD_VER_REL        0xCAFE1007
+#define UTAG_MOT_BUILD_CID        0xCAFE1008
+#define UTAG_MOT_BUILD_PRODUCT    0xCAFE1009
+#define UTAG_BUILD_FINGERPRINT    0xCAFE100A
+#define UTAG_KERNEL_VERSION       0xCAFE1020
+
 extern struct utag *thaw_tags(size_t block_size, void *buf,
 			      enum utag_error *status);
 extern void *freeze_tags(size_t block_size, const struct utag *tags,
-- 
2.11.0

